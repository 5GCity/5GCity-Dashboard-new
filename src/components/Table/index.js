/**
 * Table Component
 * Please write a description
 *
 * @author Your Name <gpatriarca@ubiwhere.com>
 */
import React from 'react'
import styled from 'styled-components'
import { Icon } from 'element-react';
import Progress from 'components/Progress';


export default ({ children, toogleAction, ...props }) => (
  <Wrapper {...props}>
    <Header>
      {props.columns.map( (el) =>
        <Column size={el.size} key={el.title}>{el.title}</Column>
      )}
      <Column></Column>
    </Header>
    <Container>
      {props.data && props.data.map((item) => {
        return[
        <Row key={item.id} onClick={() => toogleAction(item)}>
          {props.columns && props.columns.map(({ prop, render, size }) => {
            const colProps = { size }
            return [
              render && item[prop] && <Column key={item.id} {...colProps}> <ColumnTitle>{render(item[prop])}</ColumnTitle> <ProgressContainer percentage={item[prop].percentage} type={'line'} textInside={false} showText={false} status={item[prop].status} /></Column>,
              !render && item[prop] && <Column key={item.id} {...colProps}>{item[prop]}</Column>
            ]
          })}
          <Column key={item.id}>
            { item.open &&
              <Icon name={'arrow-up'} />
            }
            { !item.open &&
              <Icon name={'arrow-right'} />
            }
          </Column>
        </Row>,
        item.details && item.details.map((detail) =>
        <DetailRow key={detail.id} show={item.open}>
         {props.columns && props.columns.map(({ prop, render, size }) => {
            const colProps = { size }
            return [
              render && detail[prop] &&
              <Column key={detail.id} {...colProps}>
                <ColumnTitle>{render(detail[prop])}</ColumnTitle>
                <ProgressContainer
                  percentage={item[prop].percentage}
                  type={'line'}
                  textInside={false}
                  showText={false}
                  status={item[prop].status}
                />
              </Column>,
              !render && detail[prop] &&
              <Column key={detail.id} {...colProps}>
              {detail[prop]}</Column>
            ]
          })}
            <Column key={detail.id}></Column>
        </DetailRow>
        )
        ]})}
    </Container>
  </Wrapper>
)


const Wrapper = styled.div`
`

const Header = styled.div`
  display: flex;
  align-items: center;
	height: 48px;
  padding 25px 26px;
  color: #89979F;
  font-family: ${({ theme }) => theme.fontDinCondensed };
  font-size: 12px;
  letter-spacing: 1px;
	background-color: #37474F;
	box-shadow: inset 0 -1px 0 0 #47565F;
`

const Container = styled.div``

const Row = styled.div`
  display: flex;
  align-items: center;
  padding 25px 26px;
	height: 64px;
  background-color: #37474F;
  color: #FFFFFF;
  font-family: "Open Sans";
  font-size: 14px;
	box-shadow: inset 0 1px 0 0 #47565F, inset 0 -1px 0 0 #47565F;
`

const DetailRow = styled.div`
  display: flex;
  align-items: center;
  height: 64px;
  padding 25px 26px;
  background-color: #324148;
  color: #FFFFFF;
  font-family: "Open Sans";
  font-size: 14px;
  box-shadow: inset 0 1px 0 0 #47565F, inset 0 -1px 0 0 #47565F;

  ${({ show }) => !show && `
    display: none;
    height: 0px;
  `}
`

const Column = styled.div`
  align-items: center;

  :last-child {
    height: 14px;
    width: 14px;
  }

  ${({ size }) => size && `
    flex-basis: ${size}%;

  `}
`
const ColumnTitle = styled.p`
  margin:16px 0px 12px 0px;
`

const ProgressContainer = styled(Progress)`
  padding-right: 160px;
  margin-bottom: 16px;
`
