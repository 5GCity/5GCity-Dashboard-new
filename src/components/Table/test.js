/**
 * Table Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Table from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const Columns = [{
    prop: 'name',
    size: 20,
    title: "Slice name"
  },
  {
    prop: 'processor',
    size: 20,
    title: "Processor",
    render: ({ total, used, type }) => {
      return `${used}/${total} ${type}`
    }
  },
  {
    prop: 'ram',
    size: 20,
    title: "RAM",
    render: ({ total, used, type }) => {
      return `${used}/${total} ${type}`
    }
  },
  {
    prop: 'bandwidth',
    size: 20,
    title: "Bandwidth",
    render: ({ total, used, type }) => {
      return `${used}/${total} ${type}`
    }
  },
  {
    prop: 'storage',
    size: 20,
    title: "Storage",
    render: ({ total, used, type }) => {
      return `${used}/${total} ${type}`
    }
  }]


  const fakeData = [
  {
    id: 1,
    name: 'Slice name',
    processor: {
      total: 2,
      used: 1,
      type :"CPU's"
    },
    ram: {
      total: 64,
      used: 12,
      type: 'GB'
    },
    bandwidth: {
      total: 500,
      used: 300,
      type: 'mbps'
    },
    storage: {
      total: 512,
      used: 330,
      type: 'GB'
    },
    details: [
      {
        id: 1,
        name: "Network service name",
        processor: {
          total: 2,
          used: 1,
          type: "CPU's"
        },
        ram: {
          total: 64,
          used: 12,
          type: "GB"
        },
        bandwidth: {
          total: 500,
          used: 300,
          type: "mbps"
        },
        storage: {
          total: 512,
          used: 330,
          type: "GB"
        }
      }
    ]
  },
  {
    id: 2,
    name: 'Slice name',
    processor: {
      total: 2,
      used: 1,
      type :"CPU's"
    },
    ram: {
      total: 64,
      used: 12,
      type: 'GB'
    },
    bandwidth: {
      total: 500,
      used: 300,
      type: 'mbps'
    },
    storage: {
      total: 512,
      used: 330,
      type: 'GB'
    }
  },
  {
    id: 3,
    name: 'Slice name',
    processor: {
      total: 4,
      used: 3,
      type :"CPU's"
    },
    ram: {
      total: 16,
      used: 12,
      type: 'GB'
    },
    bandwidth: {
      total: 700,
      used: 1000,
      type: 'mbps'
    },
    storage: {
      total: 1024,
      used: 512,
      type: 'GB'
    },
    details: [
      {
        id: 1,
        name: "Network service name",
        processor: {
          total: 2,
          used: 1,
          type: "CPU's"
        },
        ram: {
          total: 8,
          used: 4,
          type: "GB"
        },
        bandwidth: {
          total: 200,
          used: 300,
          type: "mbps"
        },
        storage: {
          total: 512,
          used: 480,
          type: "GB"
        }
      },
      {
        id: 2,
        name: "Network service name",
        processor: {
          total: 2,
          used: 2,
          type: "CPU's"
        },
        ram: {
          total: 8,
          used: 8,
          type: "GB"
        },
        bandwidth: {
          total: 500,
          used: 500,
          type: "mbps"
        },
        storage: {
          total: 512,
          used: 256,
          type: "GB"
        }
      }
    ]
    },
    {
    id: 4,
    name: 'Slice name',
    processor: {
      total: 2,
      used: 1,
      type: "CPU's"
    },
    ram: {
      total: 64,
      used: 12,
      type: 'GB'
    },
    bandwidth: {
      total: 500,
      used: 300,
      type: 'mbps'
    },
    storage: {
      total: 512,
      used: 330,
      type: 'GB'
    }
  },
  {
    id: 5,
    name: 'Slice name',
    processor: {
      total: 2,
      used: 1,
      type: "CPU's"
    },
    ram: {
      total: 64,
      used: 12,
      type: 'GB'
    },
    bandwidth: {
      total: 500,
      used: 300,
      type: 'mbps'
    },
    storage: {
      total: 512,
      used: 330,
      type: 'GB'
    }
  }
  ]

  const tree = renderer
    .create(<Table columns={Columns} data={fakeData}/>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
