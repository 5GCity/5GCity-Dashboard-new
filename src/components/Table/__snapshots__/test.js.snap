// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 48px;
  padding: 25px 26px;
  color: #89979F;
  font-size: 12px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  background-color: #37474F;
  box-shadow: inset 0 -1px 0 0 #47565F;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-basis: 20%;
  -ms-flex-preferred-size: 20%;
  flex-basis: 20%;
}

.c1:last-child {
  height: 14px;
  width: 14px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:last-child {
  height: 14px;
  width: 14px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 25px 26px;
  height: 64px;
  background-color: #37474F;
  color: #FFFFFF;
  font-family: "Open Sans";
  font-size: 14px;
  box-shadow: inset 0 1px 0 0 #47565F,inset 0 -1px 0 0 #47565F;
}

.c4 {
  margin: 16px 0px 12px 0px;
}

.c5 {
  padding-right: 160px;
  margin-bottom: 16px;
}

.c6 {
  witdh: 100%;
}

.c6 .el-progress-bar__inner {
  background-color: #8CC14E;
  border-radius: 0px;
}

.c6 .el-progress-bar__outer {
  border-radius: 0px;
  background-color: rgba(137,151,159,0.15);
  height: 8px !important;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 64px;
  padding: 25px 26px;
  background-color: #324148;
  color: #FFFFFF;
  font-family: "Open Sans";
  font-size: 14px;
  box-shadow: inset 0 1px 0 0 #47565F,inset 0 -1px 0 0 #47565F;
  display: none;
  height: 0px;
}

<div
  className=""
  data={
    Array [
      Object {
        "bandwidth": Object {
          "total": 500,
          "type": "mbps",
          "used": 300,
        },
        "details": Array [
          Object {
            "bandwidth": Object {
              "total": 500,
              "type": "mbps",
              "used": 300,
            },
            "id": 1,
            "name": "Network service name",
            "processor": Object {
              "total": 2,
              "type": "CPU's",
              "used": 1,
            },
            "ram": Object {
              "total": 64,
              "type": "GB",
              "used": 12,
            },
            "storage": Object {
              "total": 512,
              "type": "GB",
              "used": 330,
            },
          },
        ],
        "id": 1,
        "name": "Slice name",
        "processor": Object {
          "total": 2,
          "type": "CPU's",
          "used": 1,
        },
        "ram": Object {
          "total": 64,
          "type": "GB",
          "used": 12,
        },
        "storage": Object {
          "total": 512,
          "type": "GB",
          "used": 330,
        },
      },
      Object {
        "bandwidth": Object {
          "total": 500,
          "type": "mbps",
          "used": 300,
        },
        "id": 2,
        "name": "Slice name",
        "processor": Object {
          "total": 2,
          "type": "CPU's",
          "used": 1,
        },
        "ram": Object {
          "total": 64,
          "type": "GB",
          "used": 12,
        },
        "storage": Object {
          "total": 512,
          "type": "GB",
          "used": 330,
        },
      },
      Object {
        "bandwidth": Object {
          "total": 700,
          "type": "mbps",
          "used": 1000,
        },
        "details": Array [
          Object {
            "bandwidth": Object {
              "total": 200,
              "type": "mbps",
              "used": 300,
            },
            "id": 1,
            "name": "Network service name",
            "processor": Object {
              "total": 2,
              "type": "CPU's",
              "used": 1,
            },
            "ram": Object {
              "total": 8,
              "type": "GB",
              "used": 4,
            },
            "storage": Object {
              "total": 512,
              "type": "GB",
              "used": 480,
            },
          },
          Object {
            "bandwidth": Object {
              "total": 500,
              "type": "mbps",
              "used": 500,
            },
            "id": 2,
            "name": "Network service name",
            "processor": Object {
              "total": 2,
              "type": "CPU's",
              "used": 2,
            },
            "ram": Object {
              "total": 8,
              "type": "GB",
              "used": 8,
            },
            "storage": Object {
              "total": 512,
              "type": "GB",
              "used": 256,
            },
          },
        ],
        "id": 3,
        "name": "Slice name",
        "processor": Object {
          "total": 4,
          "type": "CPU's",
          "used": 3,
        },
        "ram": Object {
          "total": 16,
          "type": "GB",
          "used": 12,
        },
        "storage": Object {
          "total": 1024,
          "type": "GB",
          "used": 512,
        },
      },
      Object {
        "bandwidth": Object {
          "total": 500,
          "type": "mbps",
          "used": 300,
        },
        "id": 4,
        "name": "Slice name",
        "processor": Object {
          "total": 2,
          "type": "CPU's",
          "used": 1,
        },
        "ram": Object {
          "total": 64,
          "type": "GB",
          "used": 12,
        },
        "storage": Object {
          "total": 512,
          "type": "GB",
          "used": 330,
        },
      },
      Object {
        "bandwidth": Object {
          "total": 500,
          "type": "mbps",
          "used": 300,
        },
        "id": 5,
        "name": "Slice name",
        "processor": Object {
          "total": 2,
          "type": "CPU's",
          "used": 1,
        },
        "ram": Object {
          "total": 64,
          "type": "GB",
          "used": 12,
        },
        "storage": Object {
          "total": 512,
          "type": "GB",
          "used": 330,
        },
      },
    ]
  }
>
  <div
    className="c0"
  >
    <div
      className="c1"
      size={20}
    >
      Slice name
    </div>
    <div
      className="c1"
      size={20}
    >
      Processor
    </div>
    <div
      className="c1"
      size={20}
    >
      RAM
    </div>
    <div
      className="c1"
      size={20}
    >
      Bandwidth
    </div>
    <div
      className="c1"
      size={20}
    >
      Storage
    </div>
    <div
      className="c2"
    />
  </div>
  <div
    className=""
  >
    <div
      className="c3"
      onClick={[Function]}
    >
      <div
        className="c1"
        size={20}
      >
        Slice name
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          1/2 CPU's
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          12/64 GB
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          300/500 mbps
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          330/512 GB
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
      >
        <i
          className="el-icon-arrow-right"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="c7"
    >
      <div
        className="c1"
        size={20}
      >
        Network service name
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          1/2 CPU's
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          12/64 GB
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          300/500 mbps
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          330/512 GB
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
      />
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div
        className="c1"
        size={20}
      >
        Slice name
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          1/2 CPU's
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          12/64 GB
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          300/500 mbps
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          330/512 GB
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
      >
        <i
          className="el-icon-arrow-right"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div
        className="c1"
        size={20}
      >
        Slice name
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          3/4 CPU's
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          12/16 GB
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          1000/700 mbps
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          512/1024 GB
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
      >
        <i
          className="el-icon-arrow-right"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="c7"
    >
      <div
        className="c1"
        size={20}
      >
        Network service name
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          1/2 CPU's
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          4/8 GB
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          300/200 mbps
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          480/512 GB
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
      />
    </div>
    <div
      className="c7"
    >
      <div
        className="c1"
        size={20}
      >
        Network service name
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          2/2 CPU's
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          8/8 GB
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          500/500 mbps
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
        <p
          className="c4"
        >
          256/512 GB
        </p>
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
      />
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div
        className="c1"
        size={20}
      >
        Slice name
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          1/2 CPU's
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          12/64 GB
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          300/500 mbps
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          330/512 GB
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
      >
        <i
          className="el-icon-arrow-right"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div
        className="c1"
        size={20}
      >
        Slice name
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          1/2 CPU's
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          12/64 GB
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          300/500 mbps
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c1"
        size={20}
      >
         
        <p
          className="c4"
        >
          330/512 GB
        </p>
         
        <div
          className="el-progress el-progress--line el-progress--without-text c5 c6"
          style={Object {}}
        >
          <div
            className="el-progress-bar"
          >
            <div
              className="el-progress-bar__outer"
              style={
                Object {
                  "height": "6px",
                }
              }
            >
              <div
                className="el-progress-bar__inner"
                style={
                  Object {
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
      >
        <i
          className="el-icon-arrow-right"
          style={Object {}}
        />
      </div>
    </div>
  </div>
</div>
`;
